### PR TITLE
[NFC] Small updates for coding conventions, etc.

### PR DESCRIPTION
**Pull request by schweitzpgi**
Update to newer standard interfaces, such as TODO, etc.
Update to LLVM coding standard.

### CODE REVIEW

**Review by psteinfeld — APPROVED** [review]
All builds and tests correctly and looks good.

File: flang/lib/Lower/IntrinsicCall.cpp [hunk 1]
```diff
@@ -1661,11 +1660,8 @@ fir::ExtendedValue IntrinsicLibrary::outlineInExtendedWrapper(
     llvm::Optional<mlir::Type> resultType,
     llvm::ArrayRef<fir::ExtendedValue> args) {
   if (hasAbsentOptional(args)) {
-    // TODO
-    mlir::emitError(loc, "todo: cannot outline call to intrinsic " +
-                             llvm::Twine(name) +
-                             " with absent optional argument");
-    exit(1);
+    TODO(loc, "todo: cannot outline call to intrinsic " + llvm::Twine(name) +
```
Comment (sscalpone) [inline flang/lib/Lower/IntrinsicCall.cpp:1663]: Do you need the "todo:"?

File: flang/lib/Lower/IntrinsicCall.cpp [hunk 2]
```diff
@@ -1644,10 +1644,9 @@ IntrinsicLibrary::outlineInWrapper(GeneratorType generator,
     // them. Needs a better interface here. The issue is that we cannot easily
     // tell that a value is optional or not here if it is presents. And if it is
     // absent, we cannot tell what it type should be.
-    mlir::emitError(loc, "todo: cannot outline call to intrinsic " +
-                             llvm::Twine(name) +
-                             " with absent optional argument");
-    exit(1);
+    fir::emitFatalError(loc, "todo: cannot outline call to intrinsic " +
```
Comment (sscalpone) [inline flang/lib/Lower/IntrinsicCall.cpp:1647]: Why not use the TODO macro?

**Review by vdonaldson — APPROVED** [review]

File: flang/lib/Lower/IntrinsicCall.cpp (hunk 1 above)
Comment (schweitzpgi) [reply flang/lib/Lower/IntrinsicCall.cpp:1663]: No. I'll remove it.

File: flang/lib/Lower/IntrinsicCall.cpp (hunk 2 above)
Comment (schweitzpgi) [reply flang/lib/Lower/IntrinsicCall.cpp:1647]: +1

**Pull request merged**